File: ConsultoresGISGeoServiceBase/XML/XMLGetter.vb
Imports System.Xml

Namespace XML
    Public MustInherit Class XMLGetter
        ''' <summary>
        ''' Provee acceso al archivo Config.xml de forma transparente
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Protected Shared Function getXml() As XmlDocument
            Dim xml As New XmlDocument
            xml.Load(System.AppDomain.CurrentDomain.BaseDirectory & "\Config.xml")
            Return xml
        End Function

        ''' <summary>
        ''' Provee acceso al primer elemento que tenga el nombre especificado
        ''' </summary>
        ''' <param name="tagName"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        ''' <exception cref="XmlException">Si no existe el elemento arroja excepción</exception>
        Protected Shared Function getSingleXMLElement(ByVal tagName As String, Optional ByVal index As Integer = 0) As XmlElement
            Return getXml().GetElementsByTagName(tagName)(index)
        End Function

        ''' <summary>
        ''' Provee acceso a todos los elementos que tengan el nombre especificado
        ''' </summary>
        ''' <param name="tagName"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Protected Shared Function getElements(ByVal tagName As String) As XmlNodeList
            Return getXml().GetElementsByTagName(tagName)
        End Function
    End Class
End Namespace

File: GeoServicesSharp/XML/XMLGetter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace GeoServicesSharp.XML
{
    public abstract class XMLGetter
    {
        /// <summary>
        /// Provee acceso al archivo Config.xml de forma transparente
        /// </summary>
        protected static XmlDocument getXml()
        {
            XmlDocument xml = new XmlDocument();
            xml.Load(System.AppDomain.CurrentDomain.BaseDirectory + "\\Config.xml");
            return xml;
        }
        /// <summary>
        /// Provee acceso al primer elemento que tenga el nombre especificado
        /// </summary>
        /// <exception cref="XmlException">Si no existe el elemento arroja excepción</exception>
        protected static XmlElement getSingleXMLElement(string tagName, int index = 0) {
            return (XmlElement)getXml().GetElementsByTagName(tagName)[index];
        }
    }
}